provisioner/terraform/provision.go

```diff
@@ -49,21 +49,23 @@ func (s *server) Provision(stream proto.DRPCProvisioner_ProvisionStream) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	// Create a separate context for forcefull cancellation not tied to
+	// Create a separate context for forceful cancellation not tied to
 	// the stream so that we can control when to terminate the process.
 	killCtx, kill := context.WithCancel(context.Background())
 	defer kill()
 
 	// Ensure processes are eventually cleaned up on graceful
 	// cancellation or disconnect.
 	go func() {
-		<-stream.Context().Done()
+		<-ctx.Done()
```

Comment:
I need to ask a (perhaps stupid) question since we're relying on exit/hung timeouts to be predictable.

Do we have anyway to ensure that this specific context is cancelled in the event that heartbeats or updates are hanging/failing/timing out. Let's say network conditions are such that the stream doesn't die and this stream context remains open, but provisioner heartbeats to coderd are not coming through (perhaps stream writes simply hang).

Or, let's say it takes 3 minutes longer for this context to be cancelled than what hang detector is expecting. We would then be waiting 3 + 3 minutes and thus still potentially be canceling (SIGINT) the terraform apply for a minute after the job is marked as terminated.